jsonschema-hyperschema.xml

```diff
@@ -1946,13 +1945,173 @@ Link: <https://api.example.com/trees/1/nodes/456> rev=up
+<![CDATA[{
+    "rel": "collection",
+    "href": "/things",
+    "targetSchema": {"$ref": "thing-collection#"},
+    "submissionSchema": {"$ref": "#"}
+}]]>
+                    </artwork>
+                    <postamble>
+                        Here we see the "submissionSchema" indicating that we can create
+                        a single "thing" by submitting a representation (minus server-created
+                        fields such as "id") to the collection that is this link's target
+                        schema.  While we cannot, in general, make assumptions about the
+                        semantics of making a data submission request (in HTTP terms, a POST),
+                        <xref target="collectionAndItem" /> tells us that we MAY make such
+                        an assumption when the link relation is "collection", and that
+                        hyper-schema authors MUST NOT use a "collection" link if the data
+                        submission operation has semantics other than item creation.
+                    </postamble>
+                </figure>
+                <t>
+                    But what if we do not have any "thing"s yet?  We cannot get to the
+                    individual "thing" schema as there is no individual "thing" to fetch.
+                    And the "tag:rel.example.com,2017:thing" link in the entry point
+                    resource does not indicate how to create a "thing".  The "self" link
+                    requires the "id" to already exist, but it is assigned by the server.
+                    So we need to add another link to our entry point schema:
+                </t>
+                <figure>
+                    <preamble>
+                        This LDO would be added to the top-level "links" array in the entry
+                        point resource's hyper-schema.
+                    </preamble>
+                    <artwork>
+<![CDATA[{
+    "rel": "tag:rel.example.com,2017:thing-collection",
+    "href": "/things",
+    "submissionSchema": {
+        "$ref": "thing#"
+    },
+    "targetSchema": {
+        "$ref": "thing-collection#"
+    }
+}]]>
+                    </artwork>
+                </figure>
+                <t>
+                    <cref>
+                        Here we also see the "submissionSchema" to use to create a "thing",
+                        but how do we recognize it?  We can't use a "collection" link relation
+                        here, because there is no identifiable "thing" to serve as the context
+                        resource.  We could look at the submission schema, notice that it
+                        has a "collection" link and is therefore an item, and notice that
+                        its "collection" link produces the same(-ish?) URI, but that seems
+                        overly complicated.  And gets more so once pagination and filtering
+                        are introduced.
```

Comment:
I agree this is complicated. This gets back to my suggestion of using the "self" link on the "thing-collection" schema to hold `submissionSchema`.

Reply:
Again, as noted in #421 the point of this paragraph is to highlight that none of the options work very well.  It's not remotely clear to me what would.  The "self" link does not work, in my opinion, as noted in #421.